openy_gated_content.module

```diff
@@ -129,3 +130,39 @@ function _openy_gated_content_form_eventseries_live_stream_submit(&$form, FormSt
 function openy_gated_content_page_attachments(array &$attachments) {
   $attachments['#attached']['library'][] = 'openy_gated_content/styles';
 }
+
+/**
+ * Implements hook_form_FORM_ID_alter().
+ */
+function openy_gated_content_form_user_login_form_alter(&$form, FormStateInterface $form_state, $form_id) {
+  $form['#validate'][] = 'openy_gated_content_user_login_validation';
+}
+
+/**
+ * Implements hook_form_FORM_ID_alter().
+ */
+function openy_gated_content_form_user_pass_alter(&$form, FormStateInterface $form_state, $form_id) {
+  $form['#validate'][] = 'openy_gated_content_user_pass_validation';
+}
+
+/**
+ * Checks if user has related to Virtual Y role.
+ */
+function openy_gated_content_user_login_validation(array &$form, FormStateInterface $form_state) {
+  $account = \Drupal::entityTypeManager()->getStorage('user')->load($form_state->get('uid'));
+  if ($account && $account->hasRole('virtual_y_user')) {
```

Comment:
@jediandrey there will be many different virtual Y roles, that could be changed from project to project.
Our agreement that we have to check virtual_y_* (all roles that starts with virtual_y)